pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java

```diff
@@ -538,8 +539,13 @@ private static String getNormalizedFileKey(S3Object s3Object) {
     return fileKey;
   }
 
-  private void visitFiles(URI fileUri, boolean recursive, Consumer<S3Object> visitor)
-      throws IOException {
+  private static String getNormalizedFileKey(CommonPrefix commonPrefix) {
+    String prefix = commonPrefix.prefix();
+    return prefix.substring(0, prefix.length() - 1);
+  }
+
+  private void visitFiles(URI fileUri, boolean recursive, Consumer<S3Object> objectVisitor,
```

Comment:
Maybe we could add a small comment here clarifying that `CommonPrefix` represents keys that act like subdirectories (https://docs.aws.amazon.com/AmazonS3/latest/API/API_CommonPrefix.html).

Reply:
Done!